### PR TITLE
Add Django 4 compatibility

### PR DESCRIPTION
**Pull request by horthbynorthwest**
Unlike last time we did this #1 there haven't been many changes made on the base django-cron branch, so there's been nothing to merge in. I've updated the versions and the test matrix to cover django 4 testing, which are all passing.

### CODE REVIEW

File: django_cron/backends/lock/database.py [hunk 1]
```diff
@@ -5,13 +5,13 @@
...
     """
-    Locking cron jobs with database. Its good when you have not parallel run and want to make sure 2 jobs won't be
+    Locking cron jobs with database. It's good when you have not parallel run and want to make sure 2 jobs won't be
     fired at the same time - which may happened when job execution is longer that job interval.
     """
 
     @transaction.atomic
     def lock(self):
-        lock, created = CronJobLock.objects.get_or_create(job_name=self.job_name)
+        lock, created = CronJobLock.objects.select_for_update().get_or_create(job_name=self.job_name)
```
Comment (horthbynorthwest) [inline django_cron/backends/lock/database.py:14]: Probably isn't needed for us, but good future proofing, resolves an issue listed [here](https://github.com/Tivix/django-cron/issues/243)

File: django_cron/cron.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 from django.conf import settings
 
-from django_common.helper import send_mail
+from django.core.mail import send_mail
```
Comment (horthbynorthwest) [inline django_cron/cron.py:3]: Fixes [this issue](https://github.com/Tivix/django-cron/issues/245)

File: .github/workflows/django.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ jobs:
     strategy:
       max-parallel: 4
       matrix:
-        python-version: [3.8, 3.9, 3.10, 3.11]
+        python-version: [3.8, 3.9, '3.10', 3.11]
```
Comment (horthbynorthwest) [inline .github/workflows/django.yml:16]: Required due to https://github.com/actions/runner/issues/1989

File: .travis.yml
Comment (horthbynorthwest) [inline .travis.yml:1]: I *believe* this file is unused as it uses github/workflows for the test runners.

File: django_cron/backends/lock/database.py [hunk 4]
```diff
@@ -5,13 +5,13 @@
 
 class DatabaseLock(DjangoCronJobLock):
     """
-    Locking cron jobs with database. Its good when you have not parallel run and want to make sure 2 jobs won't be
+    Locking cron jobs with database. It's good when you have not parallel run and want to make sure 2 jobs won't be
```
Comment (rob-s-s) [inline django_cron/backends/lock/database.py:8]: "have not parallel run" doesn't read right? I'm not sure exactly what it means.

```suggestion
    Locking cron jobs with database. It's good when you have not run cron jobs in parallel and want to make sure 2 jobs won't be
```

**Review by aidan-stuart-leary — APPROVED** [review]

**Pull request merged**